File: src/backend/utils/AUTO_KEY_VIGENERE_CIPHER/service.py
from ..VIGENERE_CIPHER.service import encrypt_vigenere, decrypt_vigenere

# Function for key length and text length are equal but the key is followed by the text
def extend_key_auto_key_vigenere(key, text):
  extended_key = key + text
  return extended_key[:len(text)]

# Encrypt text using Auto Key Vigenere Cipher
def encrypt_auto_key_vigenere(text, key):
  return encrypt_vigenere(text, key)

# Decrypt text using Auto Key Vigenere Cipher
def decrypt_auto_key_vigenere(text, key):
  return decrypt_vigenere(text, key)

File: src/backend_rails/app/utils/AUTO_KEY_VIGENERE_CIPHER/service.rb
require_relative '../VIGENERE_CIPHER/service.rb'

class AutoKeyVigenereService
  def initialize
    @vigenere_service = VigenereService.new
  end

  def extend_key_auto_key_vigenere(key, text)
    extended_key = key + text
    extended_key[0, text.length]
  end

  def encrypt_auto_key_vigenere(text, key)
    @vigenere_service.encrypt_vigenere(text, key)
  end

  def decrypt_auto_key_vigenere(text, key)
    @vigenere_service.decrypt_vigenere(text, key)
  end
end